### PR TITLE
Implement 2D and 3D alpha_dropout

### PR DESCRIPTION
**Pull request by tippisum**
It implements per-channel alpha_dropout. It also creates corresponding function classes and unifies the process of dropout and alpha_dropout.

### CODE REVIEW

**Review by soumith — DISMISSED** [review]
please add tests in test_nn.py for the 2d and 3d variants.

**Comment by SsnL** [comment]
I would much rather having this in ATen, but unfortunately JIT blocks that currently (#9069).

@soumith Can we drop the *d thing and just call it `FeatureAlphaDropout`? I was deprecating `Dropout*d` with `FeatureDropout` in #9008 .

**Comment by apaszke** [comment]
+1 for not going with `Nd` names

**Comment by soumith** [comment]
agree, let's not introduce `*Nd` names, so this PR only has `FeatureAlphaDropout`, the rest should be removed.

File: torch/nn/_functions/dropout.py [hunk 1]
```diff
@@ -66,3 +66,64 @@ def symbolic(g, input, p=0.5, train=False, inplace=False):
...
+        else:
+            output = input.clone()
+
+        ctx.noise = cls._make_noise(input)
+        if ctx.p == 1:
+            a = 0
+            b = ctx.noise
+        else:
+            ctx.noise.bernoulli_(1 - ctx.p)
+            a = ((3.09091329228798008 * ctx.p + 1) * (1 - ctx.p)) ** (-0.5)
```
Comment (ezyang) [inline torch/nn/_functions/dropout.py]: Can you comment that this is 1.7580993408473766 ** 2? (In fact, I'd feel better if we still had alpha written out explicitly as a variable.

**Comment by ezyang** [comment]
math checks out

**Review by ezyang — APPROVED** [review]

**Pull request closed without merging**